examples/text_to_image/train_text_to_image.py

```diff
@@ -691,6 +710,40 @@ def collate_fn(examples):
                         accelerator.save_state(save_path)
                         logger.info(f"Saved state to {save_path}")
 
+                        if args.validation_prompt:
```

Comment:
We should add `validation_epochs` and generate according to that instead of generating after each loop. 
you could refer to this script to see how to do that 
https://github.com/huggingface/diffusers/blob/9213d81bd0ac7b8021222d811251fb590329080d/examples/dreambooth/train_dreambooth_lora.py#L926

Reply:
And this should be wrapped under the main process condition (`if accelerator.is_main_process:`) to handle situations for multi-GPU training.

Reply:
I'm currently using the args.checkpointing_steps.  Do we have a preference for # of epochs vs. # of global steps?  I slightly favor # of global steps, since that's how we're controlling checkpointing.

Reply:
Yeah, for validation images, we prefer epochs since conceptually it's a bit simpler to think of when the inference is going to take place.